examples/pipeline_wavernn/loss_mol.py

```diff
@@ -0,0 +1,81 @@
+import torch
+import torch.nn.functional as F
+
+# Adapted from wavenet vocoder:
+# https://github.com/r9y9/wavenet_vocoder/blob/master/wavenet_vocoder/mixture.py
+# Explain mol loss:
+# https://github.com/Rayhane-mamah/Tacotron-2/issues/155
+
+# Remove numpy dependency
+
+
+def log_sum_exp(x):
+    """ numerically stable log_sum_exp implementation that prevents overflow """
+
+    axis = len(x.size()) - 1
+    m, _ = torch.max(x, dim=axis)
+    m2, _ = torch.max(x, dim=axis, keepdim=True)
+    return m + torch.log(torch.sum(torch.exp(x - m2), dim=axis))
+
+
+def LossFn_Mol(y_hat, y, num_classes=65536, log_scale_min=None, reduce=True):
+    """ calculate the loss of mol mode"""
+
+    min_value = -32.23619130191664  # = float(np.log(1e-14))
```

Comment:
nit: how about `torch.log(1e-14).item()`?

Reply:
Good suggestion. It has been updated.